index.js

```diff
@@ -0,0 +1,231 @@
+  });
+
+  if (reverseCurrencyReateIndex !== -1) {
+    // Update currency object for reverse currency rate
+    currencyRates[reverseCurrencyReateIndex].rates[baseCurrencyNameValue] =
+      1 / Number(exchangeRate.value);
+  } else {
+    // Adding new currency object for reverse currency rate
+    currencyRates.push({
+      timestamp: setDate().timestamp,
+      base: targetCurrencyNameValue,
+      date: setDate().date,
+      rates: {
+        [baseCurrencyNameValue]: 1 / Number(exchangeRate.value),
+      },
+    });
+  }
+}
+
+// setting the first currency list
+
+function updateFirstCurrencySelectorList(newCurrency) {
+  let isNew = true;
+  currencyRates.forEach((item) => {
+    if (item.base === newCurrency) {
+      isNew = false;
+    }
+  });
+  if (isNew) {
+    const currencyName = document.createElement("option");
+    currencyName.value = newCurrency;
+    currencyName.innerText = newCurrency;
+    firstCurrencyList.appendChild(currencyName);
+  }
+}
+
+// setting the second currency list according to chosen option in first currency list
+
+firstCurrencyList.addEventListener("change", () => {
+  secondCurrencyList.innerHTML = '<option value="">Select Currency</option>';
+  const availableCurrencyRates = currencyRates.find((item) => {
+    return item.base === firstCurrencyList.value;
+  })?.rates;
+
+  for (let currency in availableCurrencyRates) {
+    const currencyName = document.createElement("option");
+    currencyName.value = currency;
+    currencyName.innerText = currency;
+    secondCurrencyList.appendChild(currencyName);
+  }
+});
+
+// Currency Converter
+
+function convert(event) {
+  event.preventDefault();
+
+  if (firstCurrencyAmount.value <= 0) {
+    ShowErrorMessage(firstCurrencyAmount, convertError, "Invalid Amount");
+  } else {
+    const firstCurrency = firstCurrencyList.value;
+    const secondCurrency = secondCurrencyList.value;
+    const rate = currencyRates.find((item) => item.base === firstCurrency)
+      .rates[secondCurrency];
+    const secondCurrencyAmount = rate * Number(firstCurrencyAmount.value);
+
+    result.innerText = `${firstCurrencyAmount.value} of ${firstCurrency} is equal to ${secondCurrencyAmount} of ${secondCurrency}`;
+  }
+}
+
+// Functions for Showing message to user
+
+function showSuccessMessage(message) {
+  messageContainer.style.display = "block";
+  messageText.innerText = message;
+  setTimeout(() => {
+    messageContainer.style.display = "none";
+  }, 3000);
+}
+
+function setYesNoQuestionBox(question) {
+  questionContainer.style.display = "block";
+  questionText.innerText = question;
+
+  return new Promise((resolve) => {
+    acceptBtn.addEventListener("click", () => {
+      questionContainer.style.display = "none";
+      resolve(true);
+    });
+    rejectBtn.addEventListener("click", () => {
+      questionContainer.style.display = "none";
+      resolve(false);
+    });
+  });
+}
+
+function ShowErrorMessage(inputElement, messageElement, message) {
```

Comment:
```suggestion
function showErrorMessage(inputElement, messageElement, message) {
```
Keep your function names in camelCase. We are not coding in C# in the course 😉